src/gui/DatabaseWidget.cpp

```diff
@@ -1520,13 +1524,34 @@ void DatabaseWidget::onGroupChanged()
 
 void DatabaseWidget::onDatabaseModified()
 {
-    if (!m_blockAutoSave && config()->get(Config::AutoSaveAfterEveryChange).toBool()) {
+    refreshSearch();
+    int autosaveDelay = m_db->metadata()->autosaveDelay() * 60 * 1000;  // min to msec for QTimer
+    bool autosaveAfterEveryChangeConfig = config()->get(Config::AutoSaveAfterEveryChange).toBool();
```

Comment:
I would move this local variable on line 1535 before usage, otherwise it is dead if the timer is active.

Reply:
This is outdated now that i made [abfb352](https://github.com/keepassxreboot/keepassxc/pull/9100/commits/abfb352eaa2fc888dc33d7756c3cd287e9fba2a8)